chore_: kleros and superchain remote lists added

saledjenic opened the pull request:
Added tokens lists:
- kleros
- superchain
- status-remote (this one is empty so far)

The code has been updated to use the remote list of token lists now.

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (49)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 213860b0 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6606/1/) | 2025-05-28 12:16:58 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250528-121404-213860-pr6606.aar) |
| :heavy_check_mark: | 213860b0 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6606/1/) | 2025-05-28 12:17:17 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250528-121405-213860-pr6606.zip) |
| :heavy_check_mark: | 213860b0 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6606/1/) | 2025-05-28 12:18:48 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250528-121409-213860-pr6606.zip) |
| :heavy_check_mark: | 213860b0 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6606/1/) | 2025-05-28 12:19:37 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250528-121411-213860-pr6606.zip) |
| :heavy_check_mark: | 213860b0 | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6606/1/) | 2025-05-28 12:20:26 | ~6 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250528-121415-213860-pr6606.zip) |
| :heavy_check_mark: | 213860b0 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6606/1/) | 2025-05-28 12:20:31 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250528-121407-213860-pr6606.zip) |
| :heavy_multiplication_x: | 213860b0 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/1/) | 2025-05-28 12:24:51 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/1//consoleText) |
| :heavy_multiplication_x: | 213860b0 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6606/1/) | 2025-05-28 12:50:37 | ~36 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6606/1//consoleText) |
| :heavy_check_mark: | 213860b0 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/2/) | 2025-05-28 13:17:38 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 30ad61a5 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6606/2/) | 2025-05-28 14:45:17 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250528-144210-30ad61-pr6606.aar) |
| :heavy_check_mark: | 30ad61a5 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6606/2/) | 2025-05-28 14:45:24 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250528-144211-30ad61-pr6606.zip) |
| :heavy_check_mark: | 30ad61a5 | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6606/2/) | 2025-05-28 14:46:39 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250528-144221-30ad61-pr6606.zip) |
| :heavy_check_mark: | 30ad61a5 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6606/2/) | 2025-05-28 14:46:52 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250528-144215-30ad61-pr6606.zip) |
| :heavy_check_mark: | 30ad61a5 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6606/2/) | 2025-05-28 14:47:37 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250528-144217-30ad61-pr6606.zip) |
| :heavy_check_mark: | 30ad61a5 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6606/2/) | 2025-05-28 14:48:37 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250528-144213-30ad61-pr6606.zip) |
| :heavy_check_mark: | 30ad61a5 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/3/) | 2025-05-28 14:51:42 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/3//consoleText) |
| :heavy_check_mark: | 30ad61a5 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6606/2/) | 2025-05-28 15:20:01 | ~37 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6606/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 1767a387 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6606/3/) | 2025-05-28 15:00:50 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250528-145742-1767a3-pr6606.zip) |
| :heavy_check_mark: | 1767a387 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6606/3/) | 2025-05-28 15:01:33 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250528-145742-1767a3-pr6606.aar) |
| :heavy_check_mark: | 1767a387 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6606/3/) | 2025-05-28 15:01:40 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250528-145746-1767a3-pr6606.zip) |
| :heavy_check_mark: | 1767a387 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6606/3/) | 2025-05-28 15:02:30 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250528-145751-1767a3-pr6606.zip) |
| :heavy_check_mark: | 1767a387 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6606/3/) | 2025-05-28 15:03:33 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250528-145747-1767a3-pr6606.zip) |
| :heavy_check_mark: | 1767a387 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6606/3/) | 2025-05-28 15:03:40 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250528-145744-1767a3-pr6606.zip) |
| :heavy_check_mark: | 1767a387 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/4/) | 2025-05-28 15:08:26 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/4//consoleText) |
| :heavy_multiplication_x: | 1767a387 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6606/3/) | 2025-05-28 15:53:57 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6606/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 68a5ccaa | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6606/4/) | 2025-05-28 20:28:57 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250528-202601-68a5cc-pr6606.aar) |
| :heavy_check_mark: | 68a5ccaa | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6606/4/) | 2025-05-28 20:29:03 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250528-202603-68a5cc-pr6606.zip) |
| :heavy_check_mark: | 68a5ccaa | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6606/4/) | 2025-05-28 20:30:45 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250528-202605-68a5cc-pr6606.zip) |
| :heavy_check_mark: | 68a5ccaa | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6606/4/) | 2025-05-28 20:30:51 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250528-202612-68a5cc-pr6606.zip) |
| :heavy_check_mark: | 68a5ccaa | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6606/4/) | 2025-05-28 20:31:30 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250528-202607-68a5cc-pr6606.zip) |
| :heavy_check_mark: | 68a5ccaa | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6606/4/) | 2025-05-28 20:33:11 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250528-202603-68a5cc-pr6606.zip) |
| :heavy_check_mark: | 68a5ccaa | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/5/) | 2025-05-28 20:36:02 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/5//consoleText) |
| :heavy_multiplication_x: | 68a5ccaa | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6606/4/) | 2025-05-28 21:01:42 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6606/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | ee98c6a0 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6606/5/) | 2025-05-28 21:13:44 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250528-211056-ee98c6-pr6606.zip) |
| :heavy_check_mark: | ee98c6a0 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6606/5/) | 2025-05-28 21:13:53 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250528-211055-ee98c6-pr6606.aar) |
| :heavy_check_mark: | ee98c6a0 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6606/5/) | 2025-05-28 21:14:57 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250528-211058-ee98c6-pr6606.zip) |
| :heavy_check_mark: | ee98c6a0 | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6606/5/) | 2025-05-28 21:15:18 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250528-211103-ee98c6-pr6606.zip) |
| :heavy_check_mark: | ee98c6a0 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6606/5/) | 2025-05-28 21:16:21 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250528-211100-ee98c6-pr6606.zip) |
| :heavy_check_mark: | ee98c6a0 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6606/5/) | 2025-05-28 21:18:12 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250528-211057-ee98c6-pr6606.zip) |
| :heavy_check_mark: | ee98c6a0 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/6/) | 2025-05-28 21:21:30 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/6//consoleText) |
| :heavy_multiplication_x: | ee98c6a0 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6606/5/) | 2025-05-28 21:46:25 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6606/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 94b3167a | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6606/6/) | 2025-05-29 07:34:43 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250529-073146-94b316-pr6606.aar) |
| :heavy_check_mark: | 94b3167a | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6606/6/) | 2025-05-29 07:34:49 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250529-073148-94b316-pr6606.zip) |
| :heavy_check_mark: | 94b3167a | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6606/6/) | 2025-05-29 07:36:28 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250529-073151-94b316-pr6606.zip) |
| :heavy_check_mark: | 94b3167a | [#6](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6606/6/) | 2025-05-29 07:36:34 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250529-073158-94b316-pr6606.zip) |
| :heavy_check_mark: | 94b3167a | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6606/6/) | 2025-05-29 07:37:26 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250529-073153-94b316-pr6606.zip) |
| :heavy_check_mark: | 94b3167a | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6606/6/) | 2025-05-29 07:38:57 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250529-073148-94b316-pr6606.zip) |
| :heavy_check_mark: | 94b3167a | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/7/) | 2025-05-29 07:42:06 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/7//consoleText) |
| :heavy_multiplication_x: | 94b3167a | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6606/6/) | 2025-05-29 08:06:52 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6606/6//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 22cc29ef | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6606/7/) | 2025-05-29 13:54:35 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250529-135201-22cc29-pr6606.zip) |
| :heavy_check_mark: | 22cc29ef | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6606/7/) | 2025-05-29 13:55:18 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250529-135203-22cc29-pr6606.zip) |
| :heavy_check_mark: | 22cc29ef | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6606/7/) | 2025-05-29 13:55:26 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250529-135159-22cc29-pr6606.aar) |
| :heavy_check_mark: | 22cc29ef | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6606/7/) | 2025-05-29 13:56:10 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250529-135205-22cc29-pr6606.zip) |
| :heavy_check_mark: | 22cc29ef | [#7](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6606/7/) | 2025-05-29 13:56:40 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250529-135209-22cc29-pr6606.zip) |
| :heavy_check_mark: | 22cc29ef | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6606/7/) | 2025-05-29 13:57:08 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250529-135202-22cc29-pr6606.zip) |
| :heavy_check_mark: | 22cc29ef | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/8/) | 2025-05-29 14:03:21 | ~11 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/8//consoleText) |
| :heavy_check_mark: | 22cc29ef | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6606/7/) | 2025-05-29 14:27:09 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6606/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 4c377bf5 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6606/8/) | 2025-05-29 18:57:22 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250529-185457-4c377b-pr6606.aar) |
| :heavy_check_mark: | 4c377bf5 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6606/8/) | 2025-05-29 18:57:36 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250529-185459-4c377b-pr6606.zip) |
| :heavy_check_mark: | 4c377bf5 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6606/8/) | 2025-05-29 18:58:34 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250529-185500-4c377b-pr6606.zip) |
| :heavy_check_mark: | 4c377bf5 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6606/8/) | 2025-05-29 18:58:46 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250529-185503-4c377b-pr6606.zip) |
| :heavy_check_mark: | 4c377bf5 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6606/8/) | 2025-05-29 18:59:32 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250529-185504-4c377b-pr6606.zip) |
| :heavy_check_mark: | 4c377bf5 | [#8](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6606/8/) | 2025-05-29 19:01:13 | ~6 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250529-185509-4c377b-pr6606.zip) |
| :heavy_check_mark: | 4c377bf5 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/9/) | 2025-05-29 19:03:23 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6606/9//consoleText) |
| :heavy_check_mark: | 4c377bf5 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6606/1/) | 2025-05-29 19:05:28 | ~10 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250529-185502-4c377b-pr6606.zip) |
| :heavy_check_mark: | 4c377bf5 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6606/8/) | 2025-05-29 19:28:58 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6606/8//consoleText) |

saledjenic commented on _assets/wallet/token-lists.json:
```diff
@@ -0,0 +1,23 @@
+  },
+  {
+    "id": "status-remote",
+    "sourceUrl": "https://raw.githubusercontent.com/status-im/status-go/refs/heads/develop/_assets/wallet/status-tokens-list.json"
```
This one is expected to be in the develop branch (until we find a better location for it).

dlipicar replied:
I cannot decide whether having this in the same repo is a good idea or not. Just for reference, we used to have a remote list of onramp providers in a separate repo in the past: https://github.com/status-im/crypto-on-ramps. The mechanism for fetching it was basically the same thing you're doing here.
I guess it makes sense from the POV of `develop` branch in status-go not being stuff we generally want to directly push to users, but I don't feel too strongly about it.

dlipicar replied:
also, potential for future codex integration I guess?

saledjenic replied:
Separate repo is a good option for sure.

saledjenic commented on services/wallet/token/token-lists/fetcher/constants.go:
```diff
@@ -1,8 +1,14 @@
 // TODO: update the link once we decide where to host the list of token lists.
 // remoteListOfTokenLists is the URL to fetch the list of token lists from. It needs to follow the schema defined below.
-// var remoteListOfTokenLists = fmt.Sprintf("https://raw.githubusercontent.com/status-im/status-go/refs/heads/release/%s/token-lists.json", version.Version())
+var RemoteListOfTokenLists = fmt.Sprintf("https://raw.githubusercontent.com/status-im/status-go/refs/heads/%s/_assets/wallet/token-lists.json", version.GitBranch())
```
This location is following the branch that is being built.

igor-sirotin replied:
@saledjenic This doesn't look like a good idea to me 🤔 

This should work similar to a git submodule. The reference should be concrete. 
I like the idea of having a separate repo with such configs, as @dlipicar proposed.

I can imaging a manual/auto test that PASSES in PR, because it refers to an unmodified `token-lists.json`, and FAILING on `develop`, because the `token-list.json` was modified there and is not somehow incompatible with the changes made in the PR. Result -> we have a broken `develop`.

dlipicar replied:
hmm but the whole point of this is to have some remote resource you can update without having to push a new release to the clients. It is by definition a reference to some changing thing and not an immutable resource.

saledjenic replied:
Having this in a separate repo is the best option for me. However, when we briefly discussed it today, we decided to put it where it's now for development.

> hmm but the whole point of this is to have some remote resource you can update without having to push a new release to the clients. It is by definition a reference to some changing thing and not an immutable resource.

With this approach it works exactly like that. You don't need to change the release build, it reads the list of token lists from the branch location, and status-remote list from the develop location.

saledjenic replied:
Again, just to be clear (actually I commented on those two lines because of that) I am still more for separate repo than keeping those files where they are now.

dlipicar replied:
Alright, to clear up expectations and see if we're all aligned:
- Unit and Functional tests shouldn't be using this URL, downloading anything or refreshing any token list, so if we do things right there shouldn't be test flakyness coming from this.
- The clients should access this to auto-update its token lists. E2E tests should ideally do it as well. We could have a special status-go nightly unit test checking that whatever token list the URL points to is valid, parsable, etc (we've got directory [tests-unit-network](https://github.com/status-im/status-go/blob/develop/tests-unit-network/README.MD) for integration tests that use the real network for this purpose, but it was never incorporated into CI for auto-runs)

Does this sound right?

igor-sirotin replied:
Sorry my comment was specifically about using referring to the current `version.GitBranch`.

I agree with everything said above 🙂

saledjenic replied:
> Unit and Functional tests shouldn't be using this URL, downloading anything or refreshing any token list, so if we do things right there shouldn't be test flakyness coming from this.

Unit tests should use mocked data, so those tests can be run even without an internet connection, while I am not sure if that's the case for functional tests (most likely not).

dlipicar commented:
@saledjenic have you run the analyzer after adding these new lists to check if there's some new collision we need to handle?

dlipicar commented on GIT_COMMIT:
was this file meant to be pushed?

saledjenic replied:
I am not sure how it ended up here.

dlipicar commented on services/wallet/token/token.go:
```diff
@@ -137,7 +138,7 @@ func (tm *Manager) Start(ctx context.Context, autoRefreshInterval time.Duration,
 
 	// For now we don't have the list of tokens lists remotely set so we're uisng the harcoded default lists. Once we have it
```
I'm guessing this comment is not valid anymore

dlipicar commented on services/wallet/token/token.go:
```diff
@@ -137,7 +138,7 @@ func (tm *Manager) Start(ctx context.Context, autoRefreshInterval time.Duration,
 
 	// For now we don't have the list of tokens lists remotely set so we're uisng the harcoded default lists. Once we have it
 	//we will just need to update the empty string with the correct URL.
-	tm.tokenLists.Start(ctx, "", autoRefreshInterval, autoRefreshCheckInterval)
```
Also, please check if we've got unit tests calling `Start` on their token.Manager instance. I'm guessing that will result in accesses to that URL, which shouldn't happen. Perhaps the URL should be a parameter set from outside.

saledjenic replied:
Ofc we have such places, but that's overcome by setting a very big refresh interval.

dlipicar replied:
Hmm alright that sounds a bit hacky, it's probably cleaner to just set the URL (along with these intervals) as a parameter to the tokenManager, but there's no need to do that in this PR.

saledjenic commented:
> @saledjenic have you run the analyzer after adding these new lists to check if there's some new collision we need to handle?

The analyzer is for the local/hardcodded lists only.

saledjenic commented:
@alaibe @igor-sirotin @dlipicar I've just pushed changes to the `token_collision_solving.go` file that includes token lists that are fetched from remote in the solving collision logic in a generic way.

saledjenic commented:
@alaibe @igor-sirotin @dlipicar the only question we have now is about these two locations:
- `https://raw.githubusercontent.com/status-im/status-go/refs/heads/develop/_assets/wallet/status-tokens-list.json` - location of the status remote list will always point to the develop branch
- `var RemoteListOfTokenLists = fmt.Sprintf("https://raw.githubusercontent.com/status-im/status-go/refs/heads/%s/_assets/wallet/token-lists.json", version.GitBranch())` - location of the list of token lists will point to the branch you're in.

Do we want to keep them like that or want a separate repo for maintaining tokens?

alaibe commented:
@saledjenic Until we have a real token list, I'd keep it here for the moment.
Even the original status token list is not a list I'd be proud of: many tokens are outdated and are being kept only for historical reasons.

As such keeping it closer to the source code where it is used makes more sense today
If we want to have a real "well maintained" status token list, at that moment, let's expose it properly

saledjenic committed:
chore_: kleros and superchain remote lists added

Added tokens lists:
- kleros
- superchain
- status-remote (this one is empty so far)

The code has been updated to use the remote list of token lists now.

saledjenic commented:
The following 2 files on those addresses will be used:
- https://prod.market.status.im/static/token-lists.json
- https://prod.market.status.im/static/status-token-list.json

@alaibe @igor-sirotin @dlipicar

igor-sirotin commented on services/wallet/token/token-lists/fetcher/constants.go:
```diff
@@ -1,8 +1,8 @@
-// var remoteListOfTokenLists = fmt.Sprintf("https://raw.githubusercontent.com/status-im/status-go/refs/heads/release/%s/token-lists.json", version.Version())
+// RemoteListOfTokenLists is the URL to fetch the list of token lists from. It needs to follow the schema defined below.
+// #nosec G101
+var RemoteListOfTokenLists = "https://prod.market.status.im/static/token-lists.json"
```
```suggestion
const RemoteListOfTokenLists = "https://prod.market.status.im/static/token-lists.json"
```
